skbuild/cmaker.py

```diff
@@ -491,6 +519,8 @@ def make(self, clargs=(), config="Release", source_dir=".", env=None):
                    shlex.split(os.environ.get("SKBUILD_BUILD_OPTIONS", "")))
         )
 
+        env = env_without_destdir(env)
```

Comment:
Instead, you would then update

```
def make(self, clargs=(), config="Release", source_dir=".", env=None):
```

to be something like this

```
@env_without_destdir
def make(self, clargs=(), config="Release", source_dir=".", env=None):
```

Reply:
I don't think this will work out well, at least not without breaking the interface. The env= parameter is forwarded to subprocess.call, which expects either None (inherit parent's environment), or a specific environment as a dict. Unless the context decorator implements dict as well, some translation is needed.

Also, it would have to quietly hijack the env= argument. I ended up writing it as a context manager though, but with an explicit with. One benefit I think is the communication of intent, it's really _just_ for the internal install target the environment variables should be ignored.